processing/src/main/java/org/apache/druid/segment/StringDimensionIndexer.java

```diff
@@ -94,13 +94,21 @@ public EncodedKeyComponent<int[]> processRowValsToUnsortedEncodedKeyComponent(@N
     if (dimValues == null) {
       final int nullId = dimLookup.getId(null);
       encodedDimensionValues = nullId == DimensionDictionary.ABSENT_VALUE_ID ? new int[]{dimLookup.add(null)} : new int[]{nullId};
+      if (nullId == DimensionDictionary.ABSENT_VALUE_ID) {
+        encodedDimensionValues = new int[]{dimLookup.add(null)};
+        dictionaryChanged = true;
+      } else {
+        encodedDimensionValues = new int[]{nullId};
+      }
```

Comment:
with the addition of these lines I think `dimLookup.add(null)` will be called more than before

seems like the `?:` version of this if was not removed?

Reply:
Removed the check